Add docker.ubuntu role

narezina opened the pull request.

Commit:
Add docker.ubuntu role

mgobec commented on roles/docker.ubuntu/tasks/main.yml:
```diff
@@ -0,0 +1,128 @@
+- name: Ensure that Ubuntu version is supported
+  fail:
+    msg: "{{ ansible_distribution_version }} is not an acceptable version of Ubuntu for this role"
+  when: ansible_lsb.id|lower == "ubuntu" and ( ansible_distribution_version|version_compare('12.04', '<') or ansible_distribution_version|version_compare('12.10', '=') )
```
This should be < 14.04

Commit:
Change supported Ubuntu versions for docker.ubuntu role

nivancevic commented on roles/docker.ubuntu/defaults/main.yml:
```diff
@@ -0,0 +1,27 @@
+docker_defaults_file_path: /etc/default/docker
+
+# Important if running Ubuntu 12.04-13.10 and ssh on a non-standard port
+ssh_port: 22
```
Too generic name. A prefix should be added?

nivancevic replied:
Also, Ubuntu versions lower than 14.04 don't need to be supported.

nivancevic commented on roles/docker.ubuntu/defaults/main.yml:
```diff
@@ -0,0 +1,27 @@
+## List of users to be added to 'docker' system group (disabled by default)
+## SECURITY WARNING: 
+## Be aware that granted users can easily get full root access on the docker host system!
+docker_group_members: []
```
User 'ubuntu' can be added to the docker system group.

nivancevic commented on roles/docker.ubuntu/tasks/main.yml:
```diff
@@ -0,0 +1,128 @@
+    - Restart docker
+  tags: proxy
+
+- name: Start docker
```
This can be removed if the docker package name is known (Ubuntu 14.04 +).

nivancevic commented on roles/docker.ubuntu/tasks/main.yml:
```diff
@@ -0,0 +1,128 @@
+    msg: "{{ ansible_distribution_version }} is not an acceptable version of Ubuntu for this role"
+  when: ansible_lsb.id|lower == "ubuntu" and ( ansible_distribution_version|version_compare('14.04', '<') )
+
+- name: Install docker package
```
Can anything be done in order to install the latest docker version?
For example, manually setting apt-get (https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-16-04) to fetch the latest version.

Commit:
Fix docker.ubuntu role as per peer review comments.

- remove unused variable in defaults
- remove unused conditions in tasks
- set variable when systemd is not detected
